src/diffusers/utils/peft_utils.py

```diff
@@ -147,7 +147,22 @@ def unscale_lora_layers(model, weight: Optional[float] = None):
                     module.set_scale(adapter_name, 1.0)
 
 
-def get_peft_kwargs(rank_dict, network_alpha_dict, peft_state_dict, is_unet=True):
+def get_peft_kwargs(
+    rank_dict,
+    network_alpha_dict,
+    peft_state_dict,
+    is_unet=True,
+    prefix=None,
+):
+    from ..loaders.lora_base import LORA_ADAPTER_METADATA_KEY
+
+    if LORA_ADAPTER_METADATA_KEY in peft_state_dict:
+        metadata = peft_state_dict[LORA_ADAPTER_METADATA_KEY]
+        if metadata:
+            if prefix is not None:
+                metadata = {k.replace(f"{prefix}.", ""): v for k, v in metadata.items()}
```

Comment:
Could `k.removeprefix(prefix)` be used here? The current code would also remove the prefix if it's actually an infix, so the change would not have the same effect. I think this is intended only for actual prefixes.

Reply:
I don't understand. The `prefix` is always meant to be actual (e.g., `transformer.r`, `transformer.lora_alpha`, etc.). So, I don't understand:

> The current code would also remove the prefix if it's actually an infix, so the change would not have the same effect.

Reply:
I mean this difference:

```python
>>> metadata = {"foo.bar": 0, "transformer.foo": 1, "bar.transformer.foo": 2}
>>> prefix = "transformer"
>>> {k.replace(f"{prefix}.", ""): v for k, v in metadata.items()}   # <= current code
{'foo.bar': 0, 'foo': 1, 'bar.foo': 2}
>>> {k.removeprefix(prefix + "."): v for k, v in metadata.items()}  # <= suggestion
{'foo.bar': 0, 'foo': 1, 'bar.transformer.foo': 2}
```

Reply:
Oh that is a fine suggestion. Thanks! Let me apply this to the codebase wide.